src/redux/reducers/componentConfig/updateConfig.js

```diff
@@ -4,7 +4,14 @@ const updateConfig = (state, action) => {
   const { plotUuid, configChange } = action.payload;
 
   const newConfig = _.cloneDeep(state[plotUuid].config);
-  _.merge(newConfig, configChange);
+
+  const mergeConfig = (objValue, srcValue) => {
+    if (_.isArray(objValue) && srcValue) {
+      return srcValue;
+    }
+  };
+
```

Comment:
why was this change necessary?

Reply:
Because before if you have a property i.e `something:['a',' b', 'c']` and you want to set it to ['a'] - the config change. The way that merge works (not an intersection) it won't overwrite the previous value of **something** to be equal to just ['a'] but instead it will remain unchanged. So this way we check if the value is an array and if the new value exists we set the previous one to the new one instead of doing a normal merge.

Reply:
I didn't understand your answer, can you give me a specific example of a case when that happens?

Reply:
if we have a config field `{something: ['a', 'b','c']}` and we want to change it to `{something:['a']}` before it was `_.merge({something: ['a', 'b','c']}, {something:['a']}) ` - this would still return` {something: ['a', 'b','c']}`, not  `{something:['a']}` (which we wanted to change it) due to how merge works https://lodash.com/docs/4.17.15  Example would be when you add and remove metadata tracks or selected genes - it was impossible to remove a selected gene before because the array would stay the same.

Reply:
thanks Stefan! I understand now